### PR TITLE
Fix vcpkg_fixup_cmake_targets to work correctly on MinGW

### PR DESCRIPTION
**Pull request by koprok**
Handle executable suffix correctly on all targets.

### CODE REVIEW

File: scripts/cmake/vcpkg_fixup_cmake_targets.cmake [hunk 1]
```diff
@@ -53,11 +53,7 @@ function(vcpkg_fixup_cmake_targets)
         set(_vfct_TARGET_PATH share/${PORT})
     endif()
 
-    if(NOT VCPKG_CMAKE_SYSTEM_NAME OR VCPKG_CMAKE_SYSTEM_NAME STREQUAL "WindowsStore")
-        set(EXECUTABLE_SUFFIX "\\.exe")
-    else()
-        set(EXECUTABLE_SUFFIX)
-    endif()
+    string(REPLACE "." "\\." EXECUTABLE_SUFFIX "${VCPKG_TARGET_EXECUTABLE_SUFFIX}")
```
Comment (JackBoosY) [inline scripts/cmake/vcpkg_fixup_cmake_targets.cmake:56]: According to the latest code, `EXECUTABLE_SUFFIX` should be empty on UNIX.

**Review by JackBoosY — CHANGES_REQUESTED** [review]

File: scripts/cmake/vcpkg_fixup_cmake_targets.cmake (hunk 1 above)
Comment (koprok) [reply scripts/cmake/vcpkg_fixup_cmake_targets.cmake:56]: Isn't it exactly like this after my change? On UNIX `${VCPKG_TARGET_EXECUTABLE_SUFFIX}` is empty, so after replacing "." with "\\." nothing happens and `EXECUTABLE_SUFFIX` is also set to empty.

Comment (JackBoosY) [reply scripts/cmake/vcpkg_fixup_cmake_targets.cmake:56]: @koprok Hmm..make sence.

**Review by JackBoosY — CHANGES_REQUESTED** [review]
Please update the akali's version info to test the changes. See [documentation](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md#update-the-version-field-in-the-control-file-of-any-modified-ports).

**Comment by koprok** [comment]
> Please update the akali's version info to test the changes. See [documentation](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md#update-the-version-field-in-the-control-file-of-any-modified-ports).

@JackBoosY, sorry I am not very sure what you mean. Could you please clarify? Do you want me to raise akali's port version (from 1.41 to say 1.41-1) and commit this change to my branch? Why is this necessary and why exactly akali's port? Sorry for asking, I am just curious to know... for the next time I submit some similar changes? :)

**Comment by JackBoosY** [comment]
@koprok We need to bump the version of the port that uses `vcpkg_fixup_cmake_targets` to test whether the PR modification is correct.

**Comment by linquize** [comment]
Should we follow the latest convention using `Version: ` and `Port-Version: ` pair?

**Comment by koprok** [comment]
> Should we follow the latest convention using `Version: ` and `Port-Version: ` pair?

@linquize, I cannot find anything about such convention in Vcpkg documentation and I also don't see any port using it.

**Comment by linquize** [comment]
`libarchive` is using the new `Version: `and `Port-Version: `pair

**Comment by koprok** [comment]
> `libarchive` is using the new `Version: `and `Port-Version: `pair

Ahh sorry, you are right @linquize ! I was looking at my mingw-vcpkg-fixup-cmake-targets branch instead of upstream master. Should I use `Port-Version` instead of `Version`, @JackBoosY, what do you think ?

**Comment by JackBoosY** [comment]
Opencc regression is note related with this PR.
We need to wait for fix it first.

**Comment by koprok** [comment]
> Opencc regression is note related with this PR.
> We need to wait for fix it first.

Fixed in #12246

File: ports/akali/CONTROL [hunk 2]
```diff
@@ -1,4 +1,4 @@
 Source: akali
-Version: 1.41
+Version: 1.41-1
```
Comment (JackBoosY) [inline ports/akali/CONTROL:2]:
```suggestion
Version: 1.41
Port-Version: 1
```

**Review by JackBoosY — CHANGES_REQUESTED** [review]

**Review by JackBoosY — APPROVED** [review]

**Comment by JackBoosY** [comment]
Waiting for merge #12458.

**Pull request merged**